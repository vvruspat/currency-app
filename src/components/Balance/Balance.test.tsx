import { render, screen } from "@testing-library/react";
import { Balance } from ".";

test("renders balance USD", () => {
  render(
    <Balance data-testid="test-balance" balance={1234.56} currency="USD" />,
    {}
  );
  const BalanceElement = screen.getByTestId("test-balance");
  expect(BalanceElement).toBeInTheDocument();
  expect(BalanceElement).toHaveTextContent("Balance: $1,234.56");
  expect(BalanceElement).toMatchSnapshot();
});

test("renders balance EUR", () => {
  render(
    <Balance data-testid="test-balance" balance={128734.56} currency="EUR" />,
    {}
  );
  const BalanceElement = screen.getByTestId("test-balance");
  expect(BalanceElement).toBeInTheDocument();
  expect(BalanceElement).toHaveTextContent("Balance: €128,734.56");
  expect(BalanceElement).toMatchSnapshot();
});

test("renders balance GBP", () => {
  render(
    <Balance data-testid="test-balance" balance={1287} currency="GBP" />,
    {}
  );
  const BalanceElement = screen.getByTestId("test-balance");
  expect(BalanceElement).toBeInTheDocument();
  expect(BalanceElement).toHaveTextContent("Balance: £1,287.00");
  expect(BalanceElement).toMatchSnapshot();
});
