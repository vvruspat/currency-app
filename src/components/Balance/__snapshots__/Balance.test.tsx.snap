// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders balance EUR 1`] = `
<div
  data-testid="test-balance"
>
  Balance:
   
  €128,734.56
</div>
`;

exports[`renders balance GBP 1`] = `
<div
  data-testid="test-balance"
>
  Balance:
   
  £1,287.00
</div>
`;

exports[`renders balance USD 1`] = `
<div
  data-testid="test-balance"
>
  Balance:
   
  $1,234.56
</div>
`;
